certbot-ci/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -16,25 +23,78 @@ def context(request):
         integration_test_context.cleanup()
 
 
-def test_hello_1(context):
-    assert context.http_01_port
-    assert context.tls_alpn_01_port
-    try:
-        response = requests.get(context.directory_url, verify=False)
-        response.raise_for_status()
-        assert response.json()
-        response.close()
-    except urllib3.exceptions.InsecureRequestWarning:
-        pass
+def test_manual_http_auth(context):
+    """Test the HTTP-01 challenge using manual plugin."""
+    with misc.create_http_server(context.http_01_port) as webroot:
+        manual_http_hooks = misc.manual_http_hooks(webroot)
 
+        certname = context.get_domain()
+        context.certbot([
+            'certonly', '-a', 'manual', '-d', certname,
+            '--cert-name', certname,
+            '--manual-auth-hook', manual_http_hooks[0],
+            '--manual-cleanup-hook', manual_http_hooks[1],
+            '--pre-hook', 'echo wtf.pre >> "{0}"'.format(context.hook_probe),
+            '--post-hook', 'echo wtf.post >> "{0}"'.format(context.hook_probe),
+            '--deploy-hook', 'echo deploy >> "{0}"'.format(context.hook_probe)
+        ])
 
-def test_hello_2(context):
-    assert context.http_01_port
-    assert context.tls_alpn_01_port
-    try:
-        response = requests.get(context.directory_url, verify=False)
-        response.raise_for_status()
-        assert response.json()
-        response.close()
-    except urllib3.exceptions.InsecureRequestWarning:
-        pass
+    assert_hook_execution(context.hook_probe, 'deploy')
+    assert_save_renew_hook(context.config_dir, certname)
+
+
+def test_manual_dns_auth(context):
+    """Test the DNS-01 challenge using manual plugin."""
+    certname = context.get_domain('dns')
+    context.certbot([
+        '-a', 'manual', '-d', certname, '--preferred-challenges', 'dns',
+        'run', '--cert-name', certname,
+        '--manual-auth-hook', context.manual_dns_auth_hook,
+        '--manual-cleanup-hook', context.manual_dns_cleanup_hook,
+        '--pre-hook', 'echo wtf.pre >> "{0}"'.format(context.hook_probe),
+        '--post-hook', 'echo wtf.post >> "{0}"'.format(context.hook_probe),
+        '--renew-hook', 'echo renew >> "{0}"'.format(context.hook_probe)
+    ])
+
+    with pytest.raises(AssertionError):
+        assert_hook_execution(context.hook_probe, 'renew')
+    assert_save_renew_hook(context.config_dir, certname)
+
+
+def test_renew_files_permissions(context):
+    """Test certificate file permissions upon renewal"""
+    certname = context.get_domain('renew')
+    context.certbot(['-d', certname])
+
+    assert_certs_count_for_lineage(context.config_dir, certname, 1)
+    assert_world_permissions(
+        join(context.config_dir, 'archive/{0}/privkey1.pem'.format(certname)), 0)
```

Comment:
More linux-specific separators that should be separate `join` arguments.

Reply:
Done.